upup/pkg/fi/cloudup/vspheretasks/attachiso.go

```diff
@@ -57,6 +94,73 @@ func (_ *AttachISO) CheckChanges(a, e, changes *AttachISO) error {
 }
 
 func (_ *AttachISO) RenderVC(t *vsphere.VSphereAPITarget, a, e, changes *AttachISO) error {
-	glog.Info("AttachISO.RenderVC invoked!")
+	startupScript, err := changes.BootstrapScript.ResourceNodeUp(changes.IG)
+	startupStr, err := startupScript.AsString()
+	if err != nil {
+		return fmt.Errorf("error rendering startup script: %v", err)
+	}
+	dir, err := ioutil.TempDir("", *changes.VM)
+	defer os.RemoveAll(dir)
+
+	isoFile := createISO(changes, startupStr, dir)
+	err = t.Cloud.UploadAndAttachISO(changes.VM, isoFile)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func createUserData(startupStr string, dir string) {
+	// Update the startup script to add the extra spaces for
+	// indentation when copied to the user-data file.
+	strArray := strings.Split(startupStr, "\n")
+	for i, str := range strArray {
+		if len(str) > 0 {
+			strArray[i] = "       " + str
+		}
+	}
+	startupStr = strings.Join(strArray, "\n")
+
+	data := strings.Replace(userData, "$SCRIPT", startupStr, -1)
+	userDataFile := filepath.Join(dir, "user-data")
+	glog.V(4).Infof("User data file content: %s", data)
+
+	if err := ioutil.WriteFile(userDataFile, []byte(data), 0644); err != nil {
+		glog.Fatalf("Unable to write user-data into file %s", userDataFile)
```

Comment:
Errors from createUserData and createMetaData are not getting propagated to createISO. How do you plan on handling them?